### ISSUE TITLE
gt_detail & gt_matte

### ISSUE THREAD

**Issue by npapargyr** [ZHKKKe/MODNet#69]
Hi there,

gt_detail and gt_matte are actually the same?

I am asking because they are both calculated based on the trimap and whenever I plot gt_detail and gt_matte, they are the same.

Should it be this way?

Thanks in advance!

**Comment by ZHKKKe** [comment]
Hi, Thanks for your attention.

Yes, if you plot `gt_detail` and `gt_matte`, they are the same.
However, when calculating the detail loss, a boundary mask is applied to the predicted details as:
```
pred_boundary_detail = torch.where(boundaries, trimap, pred_detail)
```
Therefore, both the absolute foreground and the absolute background in `pred_boundary_detail` are the same as `gt_detail`, which means the loss only be calculated on the boundary regions, *i.e.*, transition regions. 
If you plot `pred_detail` and `pred_matte`, they should be different.

**Comment by npapargyr** [comment]
Really appreciate your response, thanks a lot and great work as well!
